packages/core/src/transform/template/template-to-typescript.ts

```diff
@@ -315,6 +320,27 @@ export function templateToTypescript(
       });
     }
 
+    function emitBinaryOperatorExpression(
+      formInfo: SpecialFormInfo,
+      node: AST.MustacheStatement | AST.SubExpression
+    ): void {
+      emit.forNode(node, () => {
+        assert(
+          node.hash.pairs.length === 0,
+          `{{${formInfo.name}}} only accepts positional parameters`
+        );
+        assert(node.params.length === 2, `{{${formInfo.name}}} requires exactly two parameters`);
```

Comment:
```suggestion
        assert(
          node.hash.pairs.length === 0,
          () => `{{${formInfo.name}}} only accepts positional parameters`
        );
        assert(
          node.params.length === 2,
          () => `{{${formInfo.name}}} requires exactly two parameters`
        );
```

This is a tiny optimization that _probably_ matters very little in the end, but by making these assertion messages thunks we can avoid doing string concatenations every time we emit a binop expression.